rust/gitxet/tests/integration_tests/test_integration_scenarios.sh

```diff
@@ -0,0 +1,516 @@
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, no data
+
+# In the first repo, create a file and push that change.
+pushd repo_a_1
+create_data_file d1.dat 1000
+cp d1.dat ../
+git add d1.dat
+git commit -a -m "Added d1.dat"
+git push origin main
+popd
+
+# Do another clone without smudging to make sure that it's actually a pointer file
+git xet clone origin_a repo_a_nosmudge --no-smudge
+assert_is_pointer_file repo_a_nosmudge/d1.dat
+
+# Attempt to pull in things to repo_2; this should create the xet commit and 
+# then the files should be the same.
+pushd repo_a_2
+git pull
+popd
+
+assert_files_equal d1.dat repo_a_2/d1.dat
+
+# STATE:
+#  origin_a: XET, D1
+#    repo_a_1: XET, D1
+#    repo_a_2: XET, D1
+#    repo_a_3: NO XET, no data
+#    repo_a_4: NO XET, no data
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, no data
+#  origin_x2: XET, no data
+
+
+# make sure that a merge works too.
+pushd repo_a_3
+echo "More text stuff" >> t3.txt
+git add t3.txt
+git commit -a -m "Added new text file."
+
+git fetch origin 
+git merge --no-edit origin/main
+popd
+
+assert_files_equal d1.dat repo_a_3/d1.dat
+
+# STATE:
+#  origin_a: XET, D1
+#    repo_a_1: XET, D1
+#    repo_a_2: XET, D1
+#    repo_a_3: XET, D1
+#    repo_a_4: NO XET, no data
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, no data
+#  origin_x2: XET, no data
+
+
+# Now, pretend that repo_a_4 actually tries to pick up the conversion on a branch from origin_c, then adds a data file, then merges in new data in origin_a, 
+# then pushes its new data to origin_a
+pushd repo_a_4
+git remote add xet_try ../origin_x1
+git fetch xet_try
+git merge --no-edit xet_try/main
+
+create_data_file d2.dat 1000
+cp d2.dat ../
+git add d2.dat
+git commit -a -m "Added d2.dat"
+# Push up to xet_try
+git push xet_try
+
+git fetch origin
+git merge --no-edit origin/main
+
+git push origin
+popd
+
+# STATE:
+#  origin_a: XET, D1 + D2
+#    repo_a_1: XET, D1
+#    repo_a_2: XET, D1
+#    repo_a_3: XET, D1
+#    repo_a_4: XET, D1 + D2
+#    repo_a_5: NO XET, no data
+#    repo_a_X1: XET, no data
+#    repo_a_X2: XET, no data
+#  origin_b1: NO XET, no data
+#  origin_b2: NO XET, no data
+#  origin_x1: XET, D2 
+#  origin_x2: XET, no data 
+
+# Now, make sure that repo_a_1 can get the new data from origin_a. 
+pushd repo_a_1
+git fetch origin
+git merge --no-edit origin/main 
+popd
+
+assert_files_equal repo_a_1/d2.dat d2.dat
+
+# Now, make sure that repo_a_2 can get the new data from origin_x1. 
+pushd repo_a_2
+git remote add xet_try ../origin_x1
+git fetch xet_try
+git merge --no-edit xet_try/main
+popd
+
+assert_files_equal repo_a_2/d2.dat d2.dat
+assert_files_equal repo_a_1/d1.dat d1.dat
```

Comment:
Do we want to check `repo_a_1/d1.dat` or `repo_a_2/d1.dat`?  I assume the latter since we are done with repo_a_1 in the above step.

Reply:
AH thank you.